### PR TITLE
Add pygmt

### PR DESCRIPTION
**Pull request by moorepants**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by moorepants** [comment]
Getting this build error on Windows:

```
Verifying transaction: ...working... done
ClobberWarning: This transaction has incompatible packages due to a shared path.
  packages: conda-forge/win-64::hdf4-4.2.13-hf8e6fe8_1003, conda-forge/win-64::hdf5-1.10.5-nompi_ha405e13_1104
  path: 'library/copying'
    self._handle = _dlopen(self._name, mode)
OSError: [WinError 126] The specified module could not be found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Miniconda\conda-bld\pygmt_1588523704817\test_tmp\run_test.py", line 2, in <module>
    import pygmt
  File "C:\Miniconda\conda-bld\pygmt_1588523704817\_test_env\lib\site-packages\pygmt\__init__.py", line 30, in <module>
    _begin()
  File "C:\Miniconda\conda-bld\pygmt_1588523704817\_test_env\lib\site-packages\pygmt\session_management.py", line 16, in begin
    with Session() as lib:
  File "C:\Miniconda\conda-bld\pygmt_1588523704817\_test_env\lib\site-packages\pygmt\clib\session.py", line 177, in __enter__
    self.create("pygmt-session")
  File "C:\Miniconda\conda-bld\pygmt_1588523704817\_test_env\lib\site-packages\pygmt\clib\session.py", line 328, in create
    restype=ctp.c_void_p,
  File "C:\Miniconda\conda-bld\pygmt_1588523704817\_test_env\lib\site-packages\pygmt\clib\session.py", line 276, in get_libgmt_func
    self._libgmt = load_libgmt()
  File "C:\Miniconda\conda-bld\pygmt_1588523704817\_test_env\lib\site-packages\pygmt\clib\loading.py", line 51, in load_libgmt
    raise GMTCLibNotFoundError(msg)
pygmt.exceptions.GMTCLibNotFoundError: Error loading the GMT shared library 'gmt_w64.dll':
[WinError 126] The specified module could not be found
Tests failed for pygmt-0.1.0-py36h779f372_0.tar.bz2 - moving package to C:\Miniconda\conda-bld\broken
WARNING conda_build.build:tests_failed(2751): Tests failed for pygmt-0.1.0-py36h779f372_0.tar.bz2 - moving package to C:\Miniconda\conda-bld\broken
set PREFIX=C:\Miniconda\conda-bld\pygmt_1588523704817\_test_env
set SRC_DIR=C:\Miniconda\conda-bld\pygmt_1588523704817\test_tmp

%SRC_DIR%>call "%SRC_DIR%\conda_test_env_vars.bat" 

%SRC_DIR%>set "CONDA_SHLVL="   && 

%SRC_DIR%>conda activate "%PREFIX%" 

(%PREFIX%) %SRC_DIR%>IF 0 NEQ 0 exit 1 
TESTS FAILED: pygmt-0.1.0-py36h779f372_0.tar.bz2

(%PREFIX%) %SRC_DIR%>"%PREFIX%\python.exe" -s "%SRC_DIR%\run_test.py" 
import: 'pygmt'

(%PREFIX%) %SRC_DIR%>IF 1 NEQ 0 exit 1 
##[error]Cmd.exe exited with code '1'.
```

File: recipes/pygmt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    # If your project compiles code (such as a C extension) then add the required compilers as separate entries here.
+    # Compilers are named 'c', 'cxx' and 'fortran'.
+    - {{ compiler('c') }}
```
Comment (moorepants) [inline recipes/pygmt/meta.yaml:20]: Don't think this is needed.

**Comment by moorepants** [comment]
Here is an open issue about the clobberwarning: https://github.com/conda-forge/hdf4-feedstock/issues/27

**Comment by moorepants** [comment]
Also, seems the same errors are reported here: https://github.com/conda-forge/gmt-feedstock/issues/90

**Comment by moorepants** [comment]
Looks like this PR may be working to address this: https://github.com/GenericMappingTools/pygmt/pull/313

Maybe we should disable the Windows builds on this feedstock for now and just open an issue to track the Windows updates.

**Comment by moorepants** [comment]
@conda-forge-admin, please rerender

**Comment by moorepants** [comment]
@weiji14 and @leouieda can you sign off on being co-maintainers of this feedstock?

Once that is done, it should be ready to ship.

**Comment by weiji14** [comment]
Thanks heaps @moorepants! Yes I'm happy to be a co-maintainer for the pygmt-feedstock.

The windows build will definitely have to wait until v0.2.0, it's still unclear what the problem is.

**Comment by moorepants** [comment]
@conda-forge/python This is ready to be merged.

File: recipes/pygmt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  # the Windows builds here can likely be enabled.
+  skip: True  # [win]
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - gmt >=6.0.0
```
Comment (nehaljwani) [inline recipes/pygmt/meta.yaml:25]: Is no upper bound safe here?

Comment (moorepants) [reply recipes/pygmt/meta.yaml:25]: Do you have a suggestion?

**Comment by moorepants** [comment]
Tagged the wrong group.

@conda-forge/help-python This is ready to be merged. Thanks.

File: recipes/pygmt/meta.yaml (hunk 2 above)
Comment (weiji14) [reply recipes/pygmt/meta.yaml:25]: Yes it should be safe. PyGMT is a Python wrapper around GMT so we want it to work with any version. That said, we could make it `>=6, <7`. I don't expect GMT7 to be out in a year or two.

Comment (moorepants) [reply recipes/pygmt/meta.yaml:25]: Happy to change it do what you want @weiji14, I'm not in the position to know what the best call is.

Comment (weiji14) [reply recipes/pygmt/meta.yaml:25]: Lets put <7 in there, just im case.

Comment (moorepants) [reply recipes/pygmt/meta.yaml]: Done.

**Comment by nehaljwani** [comment]
Thanks @moorepants !

**Comment by moorepants** [comment]
The build failed on master in this repo for some reason: https://github.com/conda-forge/staged-recipes/runs/647485189

**Comment by moorepants** [comment]
I think it hit a Github api rate limit.

**Pull request merged**